packages/dashmate/src/listr/tasks/stopNodeTaskFactory.js

```diff
@@ -34,6 +37,51 @@ export default function stopNodeTaskFactory(
           }
         },
       },
+      {
+        title: 'Check node is participating in DKG',
+        skip: (ctx) => ctx.isForce || config.get('network') === NETWORK_LOCAL,
+        task: async (ctx, task) => {
+          const rpcClient = createRpcClient({
+            port: config.get('core.rpc.port'),
+            user: config.get('core.rpc.user'),
+            pass: config.get('core.rpc.password'),
+            host: await getConnectionHost(config, 'core', 'core.rpc.host'),
+          });
+
+          const { result: blockCount } = await rpcClient.getBlockCount();
+
+          const firstWindow = [Math.floor(blockCount / 24) * 24,
+            Math.floor(blockCount / 24) * 24 + 10];
+          const secondWindow = [Math.floor(blockCount / 288) * 288,
+            Math.floor(blockCount / 288) * 288 + 42];
+
+          console.log('BlockCount', blockCount);
+          console.log(`First window [${firstWindow[0]}, ${firstWindow[1]}]`);
+          console.log(`Second window [${secondWindow[0]}, ${secondWindow[1]}]`);
+
+          const isInFirstWindow = lodash.inRange(blockCount, firstWindow[0], firstWindow[1]);
+          const isInSecondWindow = lodash.inRange(blockCount, secondWindow[0], secondWindow[1]);
+
+          if (isInFirstWindow || isInSecondWindow) {
+            console.log(`Is in first window = ${isInFirstWindow}`);
+            console.log(`Is in second window = ${isInSecondWindow}`);
+
+            const agreement = await task.prompt({
```

Comment:
What if we call it in non interactive mode?

Reply:
Probably the combination would be the best. If we pass safe, then during DKG it should throw an error.

Reply:
We had a call with @pshenmic and decided when a user does restart or stop, we throw en error if the node is in the middle of DKG. He can pass `--force` to do the command anyway. We also provide a flag `--safe` that make command wait until DKG passed.